### PR TITLE
Typescriptify crypto integration tests

### PR DESCRIPTION
**Pull request by duxovni**
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/integ/matrix-client-crypto.spec.ts [hunk 1]
```diff
@@ -32,24 +32,25 @@ import { logger } from '../../src/logger';
...
+let bobTestClient: TestClient;
 const bobUserId = "@bob:localhost";
 const bobDeviceId = "bvcxz";
 const bobAccessToken = "fewgfkuesa";
-let aliMessages;
-let bobMessages;
+let aliMessages: IContent[];
+let bobMessages: IContent[];
 
 function bobUploadsDeviceKeys() {
```
Comment (turt2live) [inline spec/integ/matrix-client-crypto.spec.ts:49]: return type

**Comment by duxovni** [comment]
Oh right, I was working on top of #2505, I'll merge in develop once that gets merged and that should fix the type errors.

File: spec/integ/matrix-client-crypto.spec.ts (hunk 1 above)
Comment (turt2live) [reply spec/integ/matrix-client-crypto.spec.ts:49]: applies throughout to other functions as well (unreachable with line comments)

File: spec/integ/matrix-client-crypto.spec.ts [hunk 2]
```diff
@@ -240,7 +234,7 @@ function bobSendsReplyMessage() {
  * @return {promise} which resolves to the ciphertext for Bob's device.
  */
 function expectAliSendMessageRequest() {
-    return expectSendMessageRequest(aliTestClient.httpBackend).then(function(content) {
+    return expectSendMessageRequest(aliTestClient.httpBackend).then(function(content: IContent) {
```
Comment (turt2live) [inline spec/integ/matrix-client-crypto.spec.ts:237]: all these inline functions should probably be converted to arrow functions, but not the end of the world.

File: spec/integ/matrix-client-crypto.spec.ts [hunk 3]
```diff
@@ -593,16 +584,16 @@ describe("MatrixClient crypto", function() {
                 };
                 bobTestClient.httpBackend.when("GET", "/sync").respond(200, syncData);
 
-                const eventPromise = new Promise((resolve, reject) => {
-                    const onEvent = function(event) {
+                const eventPromise = new Promise<MatrixEvent>((resolve) => {
+                    const onEvent = function(event: MatrixEvent) {
                         logger.log(bobUserId + " received event",
-                                    event);
+                            event);
```
Comment (turt2live) [inline spec/integ/matrix-client-crypto.spec.ts:590]: I'm surprised `event` is even on its own line?

**Review by turt2live — CHANGES_REQUESTED** [review]
overall looks good - thank you! Mostly just need to add a bunch of return types :)

**Review by turt2live — APPROVED** [review]
lgtm.

For future PRs, it's generally best to break each refactor into its own commit so the review is easier, though in this case it worked out well enough. We end up collapsing the commits down to one at merge, so they're effectively free :D

**Pull request merged**